.travis/setup-windows.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh -ex
```

Comment:
Can we keep this to the same `#!/bin/env bash -ex` or at least `#!/bin/bash -ex` as the linux builds?

Reply:
This is a posix sh script

Reply:
Right, I guess what I actually wanted to ask is to change this from POSIX sh script to a strictly bash script.

Reply:
I suppose I could, I'm just not sure that's the best choice. Using pure sh is much more portable than bash, meaning these scripts will run more places, and can later be extended to other platforms. When travis is eventually removed, for example, a good deal of common setup scripts can be unified between the 3 builds. If mac ever get's back up to speed, it will work there. It will work on the BSDs with their tsch obsession. Considering that there are no bashisms as-is, I'm not sure it's wise to allow them to sneak in in the future.

Reply:
Sure, I get that, but in practice i'm not sure I like the idea. 
The linux script is already using bashisms, bash has some nice features, and /bin/sh is not as globally the same as it should be. On Arch Linux, /bin/sh is just a symlink to /bin/bash. on MacOS `/bin/sh` is just bash. On Ubuntu, it's dash. On BSD I'm guessing it's tsch.
IMO it actually makes more sense to converge on bash, then it does converge on /bin/sh.

Reply:
Just because the binaries are different does not mean the behavior is different. POSIX sh is not some alternative shell that may be implemented in whatever manner, it is an ISO standard. Bash, Zsh, Ksh, Tsch, Dash, Ash, Mrsh, etc are all supersets of plain POSIX sh, meaning any POSIX script will run without modification in them, and in the exact same manner. When bash is invoked from a symlink to sh, it runs in a strict POSIX mode, as does Zsh. Dash is a POSIX strict shell. Bash is a bad idea, because there are new features added, and someone may be using bash 4 and need bash 5. There is also the fact that MacOS recently switched to Zsh and has deprecated bash. If any other OS that uses bash by default decides to change, bash scripts will break. POSIX sh scripts will never break.

The only convergence is towards POSIX. Using bash is *divergent*, as it excludes every other shell. The POSIX standard was established based on a shell from 1988, 32 years ago. Every embedded device, laptop, desktop, mainframe (including IBM Z/OS mainframes) has access to strictly compliant POSIX sh behavior, including Windows, which has no less than 4 ways of getting Unix-like environments.

Reply:
Alright, you've convinced me.
Do you think you could also convert the linux build to be POSIX sh compliant, and add the `set -o posix` to ensure no breakages in the future?

Reply:
Yes, at some point. Probably best for a future PR when the travis build gets addressed.